### PR TITLE
feat: add license-check-and-add to all packages. and automate adding license headers

### PR DESCRIPTION
**Pull request by htztomic**
- configure workflow to include automatically checking for license headers when opening a PR against develop and add to files if check fails
- add license-check-and-add to all packages
- add apache-license-header.txt to root
- add .licenseignorefile to ensure only certain files have a license header
Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by DaveMadsen — APPROVED** [review]
Approved with questions. I don't want you blocked, but if depending on your answers to one or more of my questions, there is likely a change to consider. Let me know if you have a few more changes to review.

File: .github/workflows/check-and-add-license-header.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        with:
+          commit_message: "docs: Auto add license headers"
+          commit_options: '--no-verify --gpg-sign'
+          commit_user_name: ${{ secrets.BOT_USER }}
+          commit_user_email: ${{ secrets.BOT_USER_EMAIL }}
+          branch: ${{ steps.branch-name.outputs.head_ref_branch }}
+
+      - name: Enable branch protection on PR branch
+        uses: octokit/request-action@v2.x
+        if: ${{ always() && github.ref_protected == true }}  # Make sure to enable branch protection even if other steps fail
```
[the remark this answers is not in the document]
Comment (htztomic) [reply .github/workflows/check-and-add-license-header.yml:77]: removing enfore_admins will not delete the branch protection

File: .licenseignorefile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+*
+
+!/src/**/*.ts
+!/src/**/*.tsx
```
[the remark this answers is not in the document]
Comment (htztomic) [reply .licenseignorefile:4]: It isn't necessary, it more so recommended for source files

File: solutions/swb-app/package.json [hunk 3]
```diff
@@ -62,6 +64,7 @@
     "eslint-plugin-import": "^2.26.0",
     "istanbul-badges-readme": "1.8.1",
     "jest": "^27.5.1",
+    "license-check-and-add": "~4.0.5",
```
[the remark this answers is not in the document]
Comment (htztomic) [reply solutions/swb-app/package.json]: changed ~ to ^

**Review by DaveMadsen — APPROVED** [review]
All of the changes/answers were exactly as I expected! Thanks for making the changes!

**Review by SanketD92 — APPROVED** [review]

**Pull request merged**